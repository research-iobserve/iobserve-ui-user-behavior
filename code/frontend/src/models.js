// @flow

import { List, Map } from "immutable";
import Record from "./types";
import type {
  ApplicationT,
  ApplicationStateT,
  PageT,
  PageStateT,
  VisitT,
  VisitStateT,
  SettingsStateT,
  LayoutT,
  StateT
} from "./types";

/**
 * Application
 */
const applicationDefaults: ApplicationT = {
  id: 0,
  name: "",
  description: "",
  custom: false
};
export class Application extends Record(applicationDefaults) {
  id: number;
  name: string;
  description: string;
  custom: boolean;
}

/**
 * Page
 */
const pageDefaults: PageT = {
  id: 0,
  name: "",
  extra: Map()
};
export class Page extends Record(pageDefaults) {
  id: number;
  name: string;
  extra: Map<string, string>;
}

/**
 * Visit
 */
const visitDefaults: VisitT = {
  id: 0,
  count: 0,
  start: new Page(),
  end: new Page(),
  action: ""
};
export class Visit extends Record(visitDefaults) {
  id: number;
  count: number;
  start: Page;
  end: Page;
  action: string;
}

/**
 * ApplicationState
 */
const applicationStateDefaults: ApplicationStateT = {
  applications: List()
};
export class ApplicationState extends Record(applicationStateDefaults) {
  applications: List<Application>;
}

/**
 * PageState
 */
const pageStateDefaults: PageStateT = {
  pages: Map()
};
export class PageState extends Record(pageStateDefaults) {
  pages: Map<number, List<Page>>;
}

/**
 * VisitState
 */
const visitStateDefaults: VisitStateT = {
  visits: Map()
};
export class VisitState extends Record(visitStateDefaults) {
  visits: Map<number, List<Visit>>;
}

/**
 * SettingsState
 */
const settingsStateDefaults: SettingsStateT = {
  layout: "breadthfirst"
};
export class SettingsState extends Record(settingsStateDefaults) {
  layout: LayoutT;
}

/**
 * State
 */
const stateDefaults: StateT = {
  application: new ApplicationState(),
  page: new PageState(),
  visit: new VisitState(),
  settings: new SettingsState()
};
export class State extends Record(stateDefaults) {
  application: ApplicationState;
  page: PageState;
  visit: VisitState;
  settings: SettingsState;
}
