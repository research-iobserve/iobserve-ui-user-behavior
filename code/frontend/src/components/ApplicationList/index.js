// @flow

import React from "react";
import { Container, Row } from "reactstrap";
import styled from "styled-components";
import type { List } from "immutable";

import { Application } from "../../models";
import ApplicationListItem from "../../components/ApplicationListItem";

const StyledApplicationList = styled.div`
  background-color: #f7f7f7;
  padding-bottom: 3rem;
  padding-top: 3rem;
`;

const StyledRow = styled(Row)`
  flex: 1;
  justify-content: space-between;
`;

type Props = {
  applications: List<Application>,
  isEditor: boolean
};

function ApplicationList(props: Props) {
  return (
    <StyledApplicationList>
      <Container>
        <StyledRow>
          {props.applications.map(application => (
            <ApplicationListItem
              key={application.get("id")}
              application={application}
              isEditor={props.isEditor}
            />
          ))}
        </StyledRow>
      </Container>
    </StyledApplicationList>
  );
}

export default ApplicationList;
