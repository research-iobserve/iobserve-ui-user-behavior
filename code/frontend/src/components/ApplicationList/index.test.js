// @flow

import React from "react";
import renderer from "react-test-renderer";
import { List } from "immutable";

import { Application } from "../../models";
import ApplicationList from "./index";

describe("<ApplicationList />", () => {
  it("should render", () => {
    const tree = renderer
      .create(<ApplicationList applications={List()} isEditor={false} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  const applications: List<Application> = List([
    new Application({ id: 0, name: "amazon.de", description: "..." }),
    new Application({ id: 1, name: "facebook.com", description: "..." }),
    new Application({ id: 2, name: "heise.de", description: "..." })
  ]);

  it("should render a list", () => {
    const tree = renderer
      .create(<ApplicationList applications={applications} isEditor={false} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
