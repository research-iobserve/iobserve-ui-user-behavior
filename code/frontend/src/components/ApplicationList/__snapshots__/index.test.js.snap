exports[`<ApplicationList /> should render 1`] = `
<div
  className="kCETdc">
  <div
    className="container">
    <div
      className="fgueNY row" />
  </div>
</div>
`;

exports[`<ApplicationList /> should render a list 1`] = `
<div
  className="kCETdc">
  <div
    className="container">
    <div
      className="fgueNY row">
      <div
        className="bTmpoo card">
        <div
          className="card-block">
          <h4
            className="card-title">
            amazon.de
          </h4>
          <p
            className="card-text">
            ...
          </p>
        </div>
        <div
          className="card-block">
          <a
            className="card-link"
            onClick={[Function]}
            style={Object {}}>
            Inspect
          </a>
        </div>
      </div>
      <div
        className="bTmpoo card">
        <div
          className="card-block">
          <h4
            className="card-title">
            facebook.com
          </h4>
          <p
            className="card-text">
            ...
          </p>
        </div>
        <div
          className="card-block">
          <a
            className="card-link"
            onClick={[Function]}
            style={Object {}}>
            Inspect
          </a>
        </div>
      </div>
      <div
        className="bTmpoo card">
        <div
          className="card-block">
          <h4
            className="card-title">
            heise.de
          </h4>
          <p
            className="card-text">
            ...
          </p>
        </div>
        <div
          className="card-block">
          <a
            className="card-link"
            onClick={[Function]}
            style={Object {}}>
            Inspect
          </a>
        </div>
      </div>
    </div>
  </div>
</div>
`;
